internal/kconfig/kconfig.go

```
@@ -263,6 +263,21 @@ func PutInteger(data []byte, integer *btf.Int, n uint64) error {
 		return fmt.Errorf("invalid boolean value: %d", n)
 	}
 
+	if len(data) < int(integer.Size) {
+		return fmt.Errorf("byte array size %d is less than the integer size %d", len(data), integer.Size)
+	}
+
+	upperBound := uint64(1<<(8*integer.Size) - 1)
+	lowerBound := uint64(math.MaxUint64)
+	if integer.Encoding == btf.Signed {
+		upperBound = uint64(1<<(8*integer.Size-1) - 1)
+		lowerBound = lowerBound - upperBound - 1
+	}
+
+	if !(n <= upperBound || lowerBound < n) {
+		return fmt.Errorf("%d exceeded the given byte range %d bytes size", n, integer.Size)
+	}
+
 	switch integer.Size {
 	case 1:
```

Instead of the `upperBound`/`lowerBound` stuff from above, what about something like:

```suggestion
	case 1:
		if integer.Encoding == btf.Signed && (n > math.MaxInt8 || n < math.MinInt8) {
			return fmt.Errorf("...")
		}
```

This makes it stupidly obvious what the intention is and avoids the subtle math. Go defines all these constants, so we should just make use of them.

---

That looks good, but math.MinInt8 is not unsigned value, so condition can be 

`(n > math.MaxInt8 && n < math.MaxUint64 - math.MaxInt8)`

And I'll change to return the error message only if it's signed conditions.